src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -213,7 +214,7 @@ private BucketEntity findBucketByNameAndCheck(String bucketName) {
     }
 
     private CatalogObjectRevisionEntity findCatalogObjectByNameAndBucketAndCheck(String bucketName, String name) {
-        CatalogObjectRevisionEntity catalogObject = catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(bucketName,
+        CatalogObjectRevisionEntity catalogObject = catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(Arrays.asList(bucketName),
```

Comment:
The called method should not do any changes to your structure? So `Collections.singletonList` might be more appropriate here

Reply:
changed to Collections.singletonList

Reply:
but i don t see the relation...Collections.singletonList  is not an immutable list, it is just a list with one element. It can still be modified